devlib/utils/android.py

```diff
@@ -31,7 +31,7 @@
 import xml.etree.ElementTree
 import zipfile
 
-from pipes import quote
+from shlex import quote
```

Comment:
`pipes.quote` is hidden and shouldn't be used; [`shlex.quote`](https://docs.python.org/3/library/shlex.html?highlight=shlex#shlex.quote) was added in Py3.3 for this purpose.

Reply:
Note: we're currently still supporting Python 2.7;  please add an import guard with fallback to importing from `pipes`.

Reply:
Indeed, under the hood of CPython, [`pipes.quote` is `shlex.quote`](https://github.com/python/cpython/blob/e42b705188271da108de42b55d9344642170aa2b/Lib/pipes.py#L65) but [it's messy even for them](https://github.com/python/cpython/blob/e42b705188271da108de42b55d9344642170aa2b/Lib/pipes.py#L64). Using `shlex.quote` is incontestably the cleanest way (as that's what the Standard says) to get the `quote` function (in Py3.3+) but I agree with you that using `pipes.quote` _works_ (until it doesn't?).

Reply:
I guess I was just lazy and did not introduce an except block for something that appeared to work in Python 3 as well, using shlex when available is definitely less confusing :)